src/app/games/yinsh/Yinsh.component.ts

```diff
@@ -0,0 +1,322 @@
+    };
+
+    constructor(messageDisplayer: MessageDisplayer) {
+        super(messageDisplayer);
+        this.availableMinimaxes = [
+            new YinshMinimax(this.rules, 'YinshMinimax'),
+        ];
+        this.showScore = true;
+        this.constructedState = this.rules.node.gamePartSlice;
+        this.constructedState.hexaBoard.allCoords().forEach((coord: Coord): void => {
+            if (this.viewInfo.caseInfo[coord.y] == null) {
+                this.viewInfo.caseInfo[coord.y] = [];
+            }
+            this.viewInfo.caseInfo[coord.y][coord.x] = {
+                coord,
+                coordinates: this.getHexaCoordsAt(coord),
+                center: this.getCenterAt(coord),
+                caseClasses: [],
+                pieceClasses: [],
+                isPiece: false,
+            };
+        });
+    }
+    public updateBoard(): void {
+        const state: YinshGameState = this.rules.node.gamePartSlice;
+        this.board = state.board;
+
+        this.cancelMoveAttempt();
+        this.updateViewInfo();
+    }
+    public updateViewInfo(): void {
+        this.constructedState.hexaBoard.allCoords().forEach((coord: Coord): void => {
+            this.viewInfo.caseInfo[coord.y][coord.x].caseClasses = this.getCaseClasses(coord);
+            this.viewInfo.caseInfo[coord.y][coord.x].pieceClasses = this.getPieceClasses(coord);
+            this.viewInfo.caseInfo[coord.y][coord.x].isPiece =
+                this.constructedState.hexaBoard.getAt(coord) !== YinshPiece.EMPTY;
+        });
+        for (const player of [Player.ZERO, Player.ONE]) {
+            this.viewInfo.sideRings[player.value] = this.constructedState.sideRings[player.value];
+        }
+        if (this.movePhase === 'INITIAL_CAPTURE' || this.movePhase === 'FINAL_CAPTURE') {
+            this.viewInfo.possibleCaptures = this.rules.getPossibleCaptures(this.constructedState);
+            this.viewInfo.selectableRings = [];
+        } else if (this.movePhase === 'INITIAL_CAPTURE_SELECT_RING' || this.movePhase === 'FINAL_CAPTURE_SELECT_RING') {
+            this.viewInfo.possibleCaptures = [];
+            this.viewInfo.selectableRings =
+                this.constructedState.hexaBoard.getRingCoords(this.constructedState.getCurrentPlayer());
+        } else {
+            this.viewInfo.possibleCaptures = [];
+            this.viewInfo.selectableRings = [];
+        }
+    }
+    public getCaseClasses(coord: Coord): string[] {
+        if (this.captured.some((c: Coord) => c.equals(coord))) {
+            return ['captured'];
+        } else if (this.moved.some((c: Coord) => c.equals(coord))) {
+            return ['moved'];
+        } else {
+            return [];
+        }
+    }
+    public getPieceClasses(coord: Coord): string[] {
+        const piece: YinshPiece = this.constructedState.hexaBoard.getAt(coord);
+        if (piece.isRing) {
+            const playerClass: string = this.getPlayerClass(piece.player);
```

Comment:
can be put before the if/else to be re-used  ^^ ?
Ce perfectionnisme autiste j'te jure :D désolé :D